src/t_hash.c

```diff
@@ -598,6 +598,32 @@ int hashZiplistValidateIntegrity(unsigned char *zl, size_t size, int deep) {
     return ret;
 }
 
+/* Return random element from a non empty hash.
+ * sdskey and sdsvalue need to be called free.
+ * sdsvalue can be NULL in which case it's not extracted. */
+void hashTypeRandomElement(robj *hashobj, unsigned long hashsize, sds *sdskey, sds *sdsvalue) {
+    if (hashobj->encoding == OBJ_ENCODING_HT) {
+        dictEntry *de = dictGetFairRandomKey(hashobj->ptr);
+        *sdskey = sdsnew((const char *)dictGetKey(de));
```

Comment:
i was able to change the code to avoid conversion from `long` to `string` and then to `double`.

but we can't use the `addHashFieldToReply` since we need to build another dict. remove some elements from it, and then use it for reply.
so we can't easily put elements from the ziplist into the other dict.

we can complicate the code further and separate between the handling of ziplist and dict in these cases, so that in the case of the dict, we'll add the existing items from the dict rather than clone them.